models/api/api.go

```diff
@@ -0,0 +1,49 @@
+package api
+
+import "time"
+
+type App struct {
+	// ID          uint      `json:"id"` // ID unused - pubkey is used as ID
+	Name        string    `json:"name"`
+	Description string    `json:"description"`
+	NostrPubkey string    `json:"nostrPubkey"`
+	CreatedAt   time.Time `json:"createdAt"`
+	UpdatedAt   time.Time `json:"updatedAt"`
+
+	LastEventAt    *time.Time `json:"lastEventAt"`
+	ExpiresAt      *time.Time `json:"expiresAt"`
+	RequestMethods []string   `json:"requestMethods"`
+	MaxAmount      int        `json:"maxAmount"`
+	BudgetUsage    int64      `json:"budgetUsage"`
+	BudgetRenewal  string     `json:"budgetRenewal"`
+}
+
+type ListAppsResponse struct {
+	Apps []App `json:"apps"`
+}
+
+type CreateAppRequest struct {
+	Name           string `json:"name"`
+	Pubkey         string `json:"pubkey"`
+	MaxAmount      string `json:"maxAmount"`
+	BudgetRenewal  string `json:"budgetRenewal"`
+	ExpiresAt      string `json:"expiresAt"`
+	RequestMethods string `json:"requestMethods"`
+	ReturnTo       string `json:"returnTo"`
```

Comment:
I don't think this should be done in go

Reply:
EDIT: I see go constructs the URL and returns it in the create response, but was not handled. I've fixed it.